### PR TITLE
404 testing

### LINKED ISSUE #47: As a visitor, when I visit these pages while not logged in, I see a 404
- [x] User/Profile Show page

- [x] User/Profile Update/Destroy page

- [x] Garden Index/"My Gardens" Page

- [x] Create Garden page

- [x] PRIVATE Garden Show Page

- [x] Garden Update/Destroy page

- [x] Sensor Show Page

- [x] Sensor New Page

- [x] Plant Show Page

- [x] Plant New Page

### PR DESCRIPTION
**Pull request by LHJE**
# 404 Testing

This PR updates these files with a test for 404:
- `garden edit spec`
- `garden index spec`
- `garden new spec`
- `users edit spec`
- `users show spec`
- `sensor show spec`
- `sensor new spec`
- `plant show spec`
- `plant new spec`

I also had to update the routes to be able to test the plants pages, and the sensor show page

The other files updated were just updating spacing/typos.

Coverage at 97.43% because of the two skipped tests

Closes #47

### CODE REVIEW

File: spec/features/gardens/index_spec.rb [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
+                                    data: [{id: "4", type: "user"}]},
+                                 sensors: {
+                                    data: []}}}
+    end
+
+    # it 'can visit public gardens page' do
+    #   visit gardens_path
+    #   expect(page).to have_content("Public Gardens")
+    # end
+  end
```
Comment (AngelaGuardia) [inline spec/features/gardens/index_spec.rb:25]: Is there a missing test here? Or is that whole describe block not needed?
```suggestion
```

Comment (AngelaGuardia) [reply spec/features/gardens/index_spec.rb:25]: Actually do we even have a garden index?

File: spec/features/gardens/index_spec.rb [hunk 2]
```diff
@@ -1,20 +1,56 @@
 require 'rails_helper'
 
 RSpec.describe 'Index Page' do
```
Comment (leahriffell) [inline spec/features/gardens/index_spec.rb:3]: Do we need this file since we removed gardens index?

**Review by leahriffell — APPROVED** [review]
Thanks for adding all of these visitor tests!

I added a comment on the gardens index spec since we removed that page. I believe it was moved to an extension ticket.

**Review by AngelaGuardia — APPROVED** [review]
Look good Luke. I think we have a test we don't need for garden_index but since it's an extension I guess it's okay to keep it empty.

There seemed to be some stuff on here from other PRs which I thought was odd. But Since the build is passion git should be okay.

File: spec/features/gardens/index_spec.rb (hunk 2 above)
Comment (LHJE) [reply spec/features/gardens/index_spec.rb]: Yeah let's get rid of it.  Dream driving can get too dreamy ;)

File: spec/features/gardens/index_spec.rb (hunk 1 above)
Comment (LHJE) [reply spec/features/gardens/index_spec.rb]: We don't! I was being hopeful haha.  I've taken it out!

**Pull request merged**